src/main/java/tr/com/srdc/cda2fhir/transform/util/impl/BundleRequest.java

```diff
@@ -0,0 +1,178 @@
+package tr.com.srdc.cda2fhir.transform.util.impl;
+
+import java.util.List;
+import java.util.ResourceBundle;
+
+import org.hl7.fhir.dstu3.model.Base;
+import org.hl7.fhir.dstu3.model.Bundle.BundleEntryComponent;
+import org.hl7.fhir.dstu3.model.Bundle.BundleEntryRequestComponent;
+import org.hl7.fhir.dstu3.model.Bundle.HTTPVerb;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Identifier;
+import org.hl7.fhir.dstu3.model.PractitionerRole;
+import org.hl7.fhir.dstu3.model.Property;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tr.com.srdc.cda2fhir.conf.Config;
+
+public class BundleRequest {
+
+	private final static Logger logger = LoggerFactory.getLogger(BundleRequest.class);
+
+	/**
+	 * Reads the configuration file, and parses out any overriding OIDs.
+	 *
+	 * @param resourceType - the resource name of the OID
+	 * @return Array of OID strings for use.
+	 */
+	private static String[] getDefinedOIDs(String resourceType) {
+
+		String[] splitOIDs = null;
+		try {
+			final ResourceBundle rb = ResourceBundle.getBundle("config");
+			String newOIDs = rb.getString(resourceType);
+			splitOIDs = newOIDs.split(",");
+		} catch (Exception e) {
+			// do not log exception.
+		}
+		return splitOIDs;
+	}
+
+	/**
+	 * Takes a bundle entry and generates the ifNotExists String.
+	 *
+	 * @param bundleEntry
+	 * @return String for application on request.
+	 */
+	public static String generateIfNoneExist(BundleEntryComponent bundleEntry) {
+
+		String ifNotExistString = "";
+
+		// all ifNoneExist strings attempt to start with an identifier.
+		Property identifierObject = bundleEntry.getResource().getNamedProperty("identifier");
+
+		// find any overriding OIDs.
+		String[] subsetOIDArray = getDefinedOIDs(bundleEntry.getResource().getResourceType().name());
+
+		if (identifierObject != null) {
```

Comment:
I think we should skip this in the event that the resource is a medication because we're only looking at coding. If a medication comes through with an identifier on it, this would make the ifDoesNotExist for the medication an identifier instead of a CD, right? That wouldn't work with my deduplication logic since medications are only deduped by CD and nothing else.

Reply:
k, it's now skipped for meds and providerRole.